### ISSUE TITLE
`AlertDialog` renders without `aria-describedby`

### ISSUE THREAD

**Issue by nami8824** [adobe/react-spectrum#9916]
### Provide a general summary of the issue here

When using `AlertDialog` (with `role="alertdialog"`), the component renders without `aria-describedby`.

According to the W3C specification, authors SHOULD use `aria-describedby` on an `alertdialog` to reference the alert message element in the dialog.
https://www.w3.org/TR/wai-aria-1.2/#alertdialog

### 🤔 Expected Behavior?

I think one of the following behaviors should be expected:
- `AlertDialog` automatically assigns an ID to the content element and sets `aria-describedby` on the dialog to reference it.
- A development warning is shown when a rendered alertdialog lacks `aria-describedby`

### 😯 Current Behavior

`AlertDialog` renders without `aria-describedby` by default. 

### 🖥️ Steps to Reproduce

https://codesandbox.io/p/sandbox/happy-saha-jtm36v?file=%2Fsrc%2FApp.js%3A26%2C1

### Version

3.31.0

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

macOS

**Comment by snowystinger** [comment]
Thanks for the issue, it looks like we should have that, we could automatically hook it up to the content slot in v3 and s2. For RAC, maybe a warning is the best we can do in development environment. We do allow the prop already in RAC, but it should be added to v3 and S2 as well probably so people can override the possibly very verbose content. For testing, it appears as though VO does not include the description in the announcement at this time, https://bugs.webkit.org/show_bug.cgi?id=282773 possibly. At least, I wasn't able to get it announcing when I was trying it out. I'm on MacOS 26.4.1.
